src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4346,11 +4334,12 @@ private void VisitDeconstructionArguments(ArrayBuilder<DeconstructionVariable> v
                     }
                     else
                     {
-                        int targetSlot = MakeSlot(variable.Expression);
                         var targetType = variable.Type;
                         var valueType = VisitOptionalImplicitConversion(rightPart, targetType, useLegacyWarnings: true, AssignmentKind.Assignment);
                         int valueSlot = MakeSlot(rightPart);
-                        TrackNullableStateForAssignment(rightPart, targetType, targetSlot, valueType, valueSlot);
+                        int slotWatermark = GetSlotWatermark();
```

Comment:
>GetSlotWatermark [](start = 44, length = 16)

Please add link to issue (here and above) for alternate method of handling cycles in self assignments.

Reply:
Added a comment and logged [[https://github.com/dotnet/roslyn/issues/33429](https://github.com/dotnet/roslyn/issues/33429)](https://github.com/dotnet/roslyn/issues/33429).

---
In reply to: [257388053](https://github.com/dotnet/roslyn/pull/33395#discussion_r257388053) [](ancestors = 257388053)